lib/matplotlib/tests/test_backend_tk.py

```diff
@@ -177,6 +177,42 @@ def test_never_update():
     # checks them.
 
 
+@pytest.mark.backend('TkAgg', skip_on_importerror=True)
+@_isolated_tk_test(success_count=1)
+def test_toolbar_button_la_mode_icon():
+    # test that icon in LA mode can be used for buttons
+    # see GH#25164
+    import tempfile
+    import warnings
+
+    from PIL import Image
+
+    import matplotlib
+    import matplotlib.pyplot as plt
+    from matplotlib.backend_tools import ToolToggleBase
+
+    # tweaking toolbar raises an UserWarning
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", UserWarning)
+        matplotlib.rcParams["toolbar"] = "toolmanager"
+
+    # create an icon in LA mode
+    with tempfile.TemporaryDirectory() as tempdir:
+        img = Image.new("LA", (26, 26))
+        tmp_img_path = os.path.join(tempdir, "test_la_icon.png")
+        img.save(tmp_img_path)
+
+        class CustomTool(ToolToggleBase):
+            image = tmp_img_path
+
+        fig = plt.figure()
+        toolmanager = fig.canvas.manager.toolmanager
+        toolbar = fig.canvas.manager.toolbar
+        toolmanager.add_tool("test", CustomTool)
+        toolbar.add_tool("test", "group")
+        print("success")
```

Comment:
stray print statement

Reply:
No because the `@_isolated_tk_test(success_count=1)` decorator looks for these print statement to mark the test as success by parsing the `stderr`. I could remove it and but `success_count=0`. Would that be preferable?

Reply:
OK, I'd never seen a test structured that way, but if you are sure it's correct, no problem.  Maybe add a comment?

Reply:
I'll get rid of the print